proposals/0000-reduce-fragmentation.md

```diff
@@ -0,0 +1,332 @@
+
+This proposal aims to allow developers to target more platforms with cross-platform APIs, and deliver better performance when targeting browsers. Features for Android, iOS, and Web are unified by aligning with the Web standard to a) minimize the overhead when running in browsers, and b) reduce developer education required to learn these features.
+
+## Detailed design
+
+Many of these additions are already in progress and have been proven in a user-space shim at Meta. The features that cannot be shimmed are explicitly called out below. Those features require core changes to React Native. Other features can be shimmed in user-space JavaScript if desired (with any associated performance cost of doing so).
+
+### New common props
+
+Certain props common to all core components have direct equivalents in React DOM. We should support those props where possible. For example, many of the existing `accessibility*` props in React Native are currently mapped to `aria-*` equivalents by React Native for Web. Supporting the ARIA interface would allow more ARIA-targeting React DOM libraries to work on React Native.
+
+* ARIA props
+  * `aria-busy` is the same as `accessibilityState.busy`.
+  * `aria-checked` is the same as `accessibilityState.checked`.
+  * `aria-disabled` is the same as `accessibilityState.disabled`.
+  * `aria-expanded` is the same as `accessibilityState.expanded`.
+  * `aria-hidden` is the same as `accessibilityElementsHidden` (iOS) / `importantforAccessibility` (Android).
+  * `aria-label` is the same as `accessibilityLabel`.
+  * `aria-live` is the same as `accessibilityLiveRegion` (with `'off'` value changed to `'none'`).
+  * `aria-modal` is the same as `accessibilityViewIsModal`.
+  * `aria-selected` is the same as `accessibilityState.selected`.
+  * `aria-valuemax` is the same as `accessibilityValue.max`.
+  * `aria-valuemin` is the same as `accessibilityValue.min`.
+  * `aria-valuenow` is the same as `accessibilityValue.now`.
+  * `aria-valuetext` is the same as `accessibilityValue.text`.
+  * `role` is the same as `accessibilityRole` (with support for web values).
+  * See prior work: [RFC Accessibility APIs](https://github.com/react-native-community/discussions-and-proposals/pull/410/files).
+* Add `dir` prop to configure element's layout / writing direction.
+* Make `id` the same as `nativeID`.
+* Make `onAuxClick` the same as React DOM `onAuxClick` PointerEvent (can't be shimmed).
+* Make `onClick` the same as React DOM `onClick` as PointerEvent (can't be shimmed).
+* Make `onPointer*` props that same as React DOM PointerEvent props (can't be shimmed).
+  * The Android and iOS implementation is currently being developed by Meta.
+* Make `tabIndex` the same as React Native `focusable` prop.
+  * Add support for `0` and `-1` values only.
+
+Example:
+
+```jsx
+<View
+  aria-hidden={false}
```

Comment:
I wonder if there are any gotchas in the infra that need to be fixed to support prop names with dashes...